src/linker/Linker/XmlDependencyRecorder.cs

```diff
@@ -83,22 +83,28 @@ public void Dispose ()
 			writer.WriteEndDocument ();
 			writer.Flush ();
 			writer.Dispose ();
-			stream.Dispose ();
+			stream?.Dispose ();
 			writer = null;
 			stream = null;
 		}
 
 		public void RecordDependency (object target, in DependencyInfo reason, bool marked)
 		{
+			if (writer == null)
+				throw new InvalidOperationException ();
```

Comment:
I think `ArgumentNullException` is appropriate here.

Reply:
`writer` isn't an argument - I'd like the exception to indicate that this method was called while the object wasn't in an appropriate state and I think InvalidOperationException is right for that.